In [2]:
import pandas as pd
import numpy as np

# Explore the Data

### Read the Data

In [3]:
crime = pd.read_csv('./crime.csv')
crime

unique_key case_number                  date                  block  \
0         9672482    HX322338  2014-06-28T11:02:00Z     030XX S NORMAL AVE   
1         8652653    HV328417  2012-06-10T12:00:00Z      029XX S LOOMIS ST   
2         9357626    HW500140  2013-10-20T02:00:00Z       029XX S CANAL ST   
3         6043894    HP146278  2008-01-27T06:42:52Z    034XX S PARNELL AVE   
4        12689202    JF230287  2022-05-04T05:05:00Z      029XX S THROOP ST   
...           ...         ...                   ...                    ...   
20077     1807040     G636924  2001-10-23T01:00:00Z  020XX N KILPATRICK AV   
20078     1859444     G701893  2001-11-21T05:00:00Z       014XX N MASON AV   
20079     1353293     G057398  2001-01-28T12:30:00Z     072XX S KIMBARK AV   
20080     1381280     G092820  2001-02-13T07:00:00Z      081XX S VERNON AV   
20081     1606451     G379049  2001-06-29T12:45:00Z     068XX S JEFFERY BL   

       iucr         primary_type  description location_description  arrest  \
0      1310      CRIMINAL DAMAGE  TO PROPERTY            APARTMENT    True   
1      1310      CRIMINAL DAMAGE  TO PROPERTY     RESIDENCE-GARAGE   False   
2      1310      CRIMINAL DAMAGE  TO PROPERTY               STREET   False   
3      1310      CRIMINAL DAMAGE  TO PROPERTY            RESIDENCE   False   
4      1310      CRIMINAL DAMAGE  TO PROPERTY            RESIDENCE   False   
...     ...                  ...          ...                  ...     ...   
20077  0910  MOTOR VEHICLE THEFT   AUTOMOBILE               STREET   False   
20078  0910  MOTOR VEHICLE THEFT   AUTOMOBILE               STREET   False   
20079  0910  MOTOR VEHICLE THEFT   AUTOMOBILE               STREET   False   
20080  0910  MOTOR VEHICLE THEFT   AUTOMOBILE               STREET   False   
20081  0910  MOTOR VEHICLE THEFT   AUTOMOBILE               STREET   False   

       domestic  ...  ward  community_area  fbi_code  x_coordinate  \
0         False  ...  11.0            60.0        14     1173453.0   
1         False  ...  11.0            60.0        14     1168580.0   
2         False  ...  11.0            60.0        14     1173766.0   
3         False  ...  11.0            60.0        14     1173190.0   
4         False  ...  11.0            60.0        14     1169930.0   
...         ...  ...   ...             ...       ...           ...   
20077     False  ...   NaN             NaN        07     1144681.0   
20078     False  ...   NaN             NaN        07     1136382.0   
20079     False  ...   NaN             NaN        07     1186170.0   
20080     False  ...   NaN             NaN        07     1180625.0   
20081     False  ...   NaN             NaN        07     1190691.0   

      y_coordinate  year            updated_on   latitude  longitude  \
0        1884797.0  2014  2018-02-10T03:50:01Z  41.839303 -87.639001   
1        1885324.0  2012  2016-02-04T06:33:39Z  41.840856 -87.656867   
2        1885503.0  2013  2018-02-10T03:50:01Z  41.841234 -87.637831   
3        1882150.0  2008  2018-02-28T03:56:25Z  41.832045 -87.640044   
4        1885355.0  2022  2022-05-11T04:49:57Z  41.840912 -87.651912   
...            ...   ...                   ...        ...        ...   
20077    1913196.0  2001  2015-08-17T03:03:40Z  41.917823 -87.743866   
20078    1909102.0  2001  2015-08-17T03:03:40Z  41.906741 -87.774455   
20079    1857257.0  2001  2015-08-17T03:03:40Z  41.763440 -87.593206   
20080    1850906.0  2001  2015-08-17T03:03:40Z  41.746141 -87.613724   
20081    1859729.0  2001  2015-08-17T03:03:40Z  41.770115 -87.576556   

                            location  
0      (41.839303195, -87.639000957)  
1       (41.840856037, -87.65686732)  
2      (41.841233568, -87.637831421)  
3       (41.83204543, -87.640044437)  
4       (41.84091182, -87.651912438)  
...                              ...  
20077  (41.917823214, -87.743866097)  
20078  (41.906741271, -87.774455458)  
20079   (41.76343982, -87.593206253)  
20080  (41.746141111, -87.613724292) 

### Info

In [4]:
crime.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20082 entries, 0 to 20081
Data columns (total 22 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   unique_key            20082 non-null  int64  
 1   case_number           20082 non-null  object 
 2   date                  20082 non-null  object 
 3   block                 20082 non-null  object 
 4   iucr                  20082 non-null  object 
 5   primary_type          20082 non-null  object 
 6   description           20082 non-null  object 
 7   location_description  20072 non-null  object 
 8   arrest                20082 non-null  bool   
 9   domestic              20082 non-null  bool   
 10  beat                  20082 non-null  int64  
 11  district              20082 non-null  int64  
 12  ward                  14048 non-null  float64
 13  community_area        14038 non-null  float64
 14  fbi_code              20082 non-null  object 
 15  x_coordinate       

Note: ward, community_area, x_coordinate, y_coordinate, latitude, longitude, location may have null value

### DF

In [5]:
chicago_crime = pd.DataFrame(crime)
chicago_crime.size, chicago_crime.shape

(441804, (20082, 22))

In [6]:
# Check the columns
chicago_crime.columns

Index(['unique_key', 'case_number', 'date', 'block', 'iucr', 'primary_type',
       'description', 'location_description', 'arrest', 'domestic', 'beat',
       'district', 'ward', 'community_area', 'fbi_code', 'x_coordinate',
       'y_coordinate', 'year', 'updated_on', 'latitude', 'longitude',
       'location'],
      dtype='object')

In [7]:
# check columns datatypes
chicago_crime.dtypes

unique_key                int64
case_number              object
date                     object
block                    object
iucr                     object
primary_type             object
description              object
location_description     object
arrest                     bool
domestic                   bool
beat                      int64
district                  int64
ward                    float64
community_area          float64
fbi_code                 object
x_coordinate            float64
y_coordinate            float64
year                      int64
updated_on               object
latitude                float64
longitude               float64
location                 object
dtype: object

In [8]:
# show the null values
null_values = chicago_crime.loc[chicago_crime.isna().any(axis=1)]
null_values

unique_key case_number                  date                  block  \
205       2876338    HJ541399  2003-08-05T08:45:00Z        021XX W 52ND ST   
217      11476318    JB474220  2008-03-16T12:00:00Z     048XX S SEELEY AVE   
282       6940140    HR320987  2009-05-14T06:05:00Z      054XX S DAMEN AVE   
735       2377111    HH678579  2002-09-28T04:37:00Z        012XX W 49TH ST   
749       2362517    HH662189  2002-09-21T12:15:00Z  016XX W GARFIELD BLVD   
...           ...         ...                   ...                    ...   
20077     1807040     G636924  2001-10-23T01:00:00Z  020XX N KILPATRICK AV   
20078     1859444     G701893  2001-11-21T05:00:00Z       014XX N MASON AV   
20079     1353293     G057398  2001-01-28T12:30:00Z     072XX S KIMBARK AV   
20080     1381280     G092820  2001-02-13T07:00:00Z      081XX S VERNON AV   
20081     1606451     G379049  2001-06-29T12:45:00Z     068XX S JEFFERY BL   

       iucr                primary_type                     description  \
205    501A               OTHER OFFENSE            ANIMAL ABUSE/NEGLECT   
217    1754  OFFENSE INVOLVING CHILDREN  AGG SEX ASSLT OF CHILD FAM MBR   
282    2017                   NARCOTICS              MANU/DELIVER:CRACK   
735    0460                     BATTERY                          SIMPLE   
749    0460                     BATTERY                          SIMPLE   
...     ...                         ...                             ...   
20077  0910         MOTOR VEHICLE THEFT                      AUTOMOBILE   
20078  0910         MOTOR VEHICLE THEFT                      AUTOMOBILE   
20079  0910         MOTOR VEHICLE THEFT                      AUTOMOBILE   
20080  0910         MOTOR VEHICLE THEFT                      AUTOMOBILE   
20081  0910         MOTOR VEHICLE THEFT                      AUTOMOBILE   

                 location_description  arrest  domestic  ...  ward  \
205                         RESIDENCE   False     False  ...  16.0   
217                         RESIDENCE    True      True  ...  15.0   
282    PARKING LOT/GARAGE(NON.RESID.)    True     False  ...  16.0   
735                            STREET   False     False  ...  20.0   
749                          SIDEWALK   False     False  ...  16.0   
...                               ...     ...       ...  ...   ...   
20077                          STREET   False     False  ...   NaN   
20078                          STREET   False     False  ...   NaN   
20079                          STREET   False     False  ...   NaN   
20080                          STREET   False     False  ...   NaN   
20081                          STREET   False     False  ...   NaN   

       community_area  fbi_code  x_coordinate y_coordinate  year  \
205              61.0        26           NaN          NaN  2003   
217              61.0        02           NaN          NaN  2008   
282              61.0        18           NaN          NaN  2009   
735              61.0       08B           NaN          NaN  2002   
749              61.0       08B           NaN          NaN  2002   
...               ...       ...           ...          ...   ...   
20077             NaN        07     1144681.0    1913196.0  2001   
20078             NaN        07     1136382.0    1909102.0  2001   
20079             NaN        07     1186170.0    1857257.0  2001   
20080             NaN        07     1180625.0    1850906.0  2001   
20081             NaN        07     1190691.0    1859729.0  2001   

                 updated_on   latitude  longitude  \
205    2015-08-17T03:03:40Z        NaN        NaN   
217    2019-01-26T04:02:53Z        NaN        NaN   
282    2015-08-17T03:03:40Z        NaN        NaN   
735    2015-08-17T03:03:40Z        NaN        NaN   
749    2015-08-17T03:03:40Z        NaN        NaN   
...                     ...        ...        ...   
20077  2015-08-17T03:03:40Z  41.917823 -87.743866   
20078  2015-08-17T03:03:40Z  41.906741 -87.774455   
20079  2015-08-17T03:03:40Z  41.76

In [9]:
# Working with non null data
data = chicago_crime.dropna()

In [10]:
# Show the crime types
data['primary_type'].value_counts()

BATTERY                              2479
THEFT                                2346
CRIMINAL DAMAGE                      2010
NARCOTICS                            1385
ASSAULT                               958
OTHER OFFENSE                         916
BURGLARY                              799
MOTOR VEHICLE THEFT                   688
DECEPTIVE PRACTICE                    528
ROBBERY                               479
CRIMINAL TRESPASS                     405
WEAPONS VIOLATION                     240
OFFENSE INVOLVING CHILDREN            142
PUBLIC PEACE VIOLATION                115
PROSTITUTION                          101
SEX OFFENSE                            51
INTERFERENCE WITH PUBLIC OFFICER       48
ARSON                                  37
CRIM SEXUAL ASSAULT                    35
GAMBLING                               30
HOMICIDE                               23
LIQUOR LAW VIOLATION                   20
KIDNAPPING                             18
STALKING                          

In [11]:
# Show crime areas
data['block'].value_counts().head(25)

038XX W ROOSEVELT RD         61
051XX S PULASKI RD           38
032XX W ROOSEVELT RD         35
039XX W ROOSEVELT RD         35
021XX E 87TH ST              35
049XX N MILWAUKEE AVE        32
002XX W GARFIELD BLVD        30
008XX E 103RD ST             29
054XX S WENTWORTH AVE        29
033XX W FILLMORE ST          27
039XX W GRENSHAW ST          26
049XX S ARCHER AVE           26
036XX W DOUGLAS BLVD         26
046XX S DAMEN AVE            25
031XX S ASHLAND AVE          25
051XX S WENTWORTH AVE        24
037XX W ROOSEVELT RD         24
012XX S INDEPENDENCE BLVD    24
038XX W GRENSHAW ST          23
054XX S DAMEN AVE            23
035XX W ROOSEVELT RD         23
045XX S DAMEN AVE            23
053XX S PULASKI RD           23
007XX E 111TH ST             23
013XX S INDEPENDENCE BLVD    22
Name: block, dtype: int64

In [12]:
# I assume some of the least crimes would be severe crimes 
chicago_crime['description'].value_counts().tail(10)

AGGRAVATED - OTHER                             1
EXTORTION                                      1
ARMED WHILE UNDER THE INFLUENCE                1
SEX OFFENDER - FAIL TO REGISTER NEW ADDRESS    1
ATTEMPT ARMED - OTHER DANGEROUS WEAPON         1
POSSESSION: SYNTHETIC MARIJUANA                1
GUN OFFENDER: DUTY TO REGISTER                 1
POSSESS - SYNTHETIC DRUGS                      1
VEHICLE TITLE / REGISTRATION OFFENSE           1
ESCAPE                                         1
Name: description, dtype: int64

In [13]:
chicago_crime['description'].value_counts().size

291

In [14]:
# check domestic and arrest
# another way: chicago_crime[chicago_crime.columns[8:10]]
chicago_crime.loc[:, ['domestic', 'arrest']]

domestic  arrest
0         False    True
1         False   False
2         False   False
3         False   False
4         False   False
...         ...     ...
20077     False   False
20078     False   False
20079     False   False
20080     False   False
20081     False   False

[20082 rows x 2 columns]

In [15]:
# Check the arrest total
# chicago_crime['arrest'] == True
count = 0
for i in chicago_crime['arrest']:
    if i == True:
        count +=1
print(count)

5215


In [16]:
# Check the domestic total
# chicago_crime['domestic'] == True
count = 0
for i in chicago_crime['domestic']:
    if i == True:
        count +=1
print(count)

2788


In [17]:
# Store the domestic and arrest columns in a dictionary 
# to use conditionals
# check where arrest and domestic are True
domestic_arrest = {
    'arrest': chicago_crime['arrest'],
    'domestic': chicago_crime['domestic']
}

type(domestic_arrest)


dict

In [ ]:
# show where the arrest and domestic are True
for key, values in domestic_arrest['arrest'].items():
    if key and values == True:
        print(key, values)

In [ ]:
# show where the arrest and domestic are True
for key, values in domestic_arrest['domestic'].items():
    if key and values == True:
        print(key, values)

In [ ]:
# variable to store the index
# where domestic and arrest are True
domestic_arrest_index_true = []

# show the index where domestic and arrest are both True 
for key1, values1 in domestic_arrest['domestic'].items(): # Get the domestic index and value
    if key1 and values1 == True: # if True
        for key2, values2 in domestic_arrest['arrest'].items(): # # Get the arrest index and value
            if key2 and values2 == True: # if True
                if key1 == key2: # both arrest and domestic index must be the same 
                    domestic_arrest_index_true.append(key1) # append the index 
                    print(key1) # index that meets the conditionals
print(domestic_arrest_index_true)

In [21]:
# check the list length
# This is the total where the crime happends domestically and the assialant(s) is/are arrested
len(domestic_arrest_index_true)

527

In [ ]:
# validate the list
count_list = 0

for key1, values1 in domestic_arrest['domestic'].items(): # Get the domestic index and value
    if key1 and values1 == True: # if True
        for key2, values2 in domestic_arrest['arrest'].items(): # # Get the arrest index and value
            if key2 and values2 == True: # if True
                if key1 == key2: # both arrest and domestic index must be the same 
                    count_list += 1
                    print(key1, values1, key2, values2)

In [23]:
# check
print(count_list)

527


Note: list and the check have the same length and values, thus we can use the list (index) for analysis

In [24]:
# Turn the list into pandas Series
index_domestic_arrest = pd.Series(domestic_arrest_index_true)
index_domestic_arrest.head()

0     12
1    103
2    151
3    158
4    160
dtype: int64

Note: I change the list to Series just to ease the analysis.
- Suppose it's easier to do this in pandas

# Variables Navigations

Since I created many variables, this list would help me to navigate throughout the variables for analysis
- data = drop all null values
- chicage_crime = DF for the dataset
- null_values = all rows that possess null values on the dataset
- index_domestic_arrest = Series where domestic and arrest are True
- domestic_arrest_index_true = index_domestic_arrest's List

In [34]:
data.head(2)

unique_key case_number                  date               block  iucr  \
0     9672482    HX322338  2014-06-28T11:02:00Z  030XX S NORMAL AVE  1310   
1     8652653    HV328417  2012-06-10T12:00:00Z   029XX S LOOMIS ST  1310   

      primary_type  description location_description  arrest  domestic  ...  \
0  CRIMINAL DAMAGE  TO PROPERTY            APARTMENT    True     False  ...   
1  CRIMINAL DAMAGE  TO PROPERTY     RESIDENCE-GARAGE   False     False  ...   

   ward  community_area  fbi_code  x_coordinate y_coordinate  year  \
0  11.0            60.0        14     1173453.0    1884797.0  2014   
1  11.0            60.0        14     1168580.0    1885324.0  2012   

             updated_on   latitude  longitude                       location  
0  2018-02-10T03:50:01Z  41.839303 -87.639001  (41.839303195, -87.639000957)  
1  2016-02-04T06:33:39Z  41.840856 -87.656867   (41.840856037, -87.65686732)  

[2 rows x 22 columns]

In [30]:
# Check the top 10 location description
data['location_description'].value_counts().head(10)

STREET                            3692
RESIDENCE                         2695
SIDEWALK                          1202
APARTMENT                         1197
PARKING LOT/GARAGE(NON.RESID.)     454
OTHER                              438
RESIDENCE-GARAGE                   340
ALLEY                              332
SCHOOL, PUBLIC, BUILDING           302
VEHICLE NON-COMMERCIAL             235
Name: location_description, dtype: int64

In [32]:
# check the primary type
data['primary_type'].value_counts().head(10)

BATTERY                2479
THEFT                  2346
CRIMINAL DAMAGE        2010
NARCOTICS              1385
ASSAULT                 958
OTHER OFFENSE           916
BURGLARY                799
MOTOR VEHICLE THEFT     688
DECEPTIVE PRACTICE      528
ROBBERY                 479
Name: primary_type, dtype: int64

In [37]:
data.columns

Index(['unique_key', 'case_number', 'date', 'block', 'iucr', 'primary_type',
       'description', 'location_description', 'arrest', 'domestic', 'beat',
       'district', 'ward', 'community_area', 'fbi_code', 'x_coordinate',
       'y_coordinate', 'year', 'updated_on', 'latitude', 'longitude',
       'location'],
      dtype='object')

- focus on battery, theft, criminal damage
- correlation with (arrest, domestic, description), (location description, location)
- year

In [52]:
result = []
for crime_type in data['primary_type']:
    if crime_type == 'BATTERY':
        result.append(crime_type)
result

['BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BATTERY',
 'BA

In [55]:
# Check
'CRIMINAL DAMAGE' in result, 'THEFT' in result, 'BATTERY' in result, 'NARCOTICS'in result, 'OTHER OFFENCE' in result, 

(False, False, True, False, False)

In [65]:
# Check BATTERY relation with description, arrest, domestic
battery_crime = data[data['primary_type'] == 'BATTERY' ]
battery_crime.shape

(2479, 22)

In [77]:
# take a look at description column
print(battery_crime['description'].value_counts().shape)
print(battery_crime['description'].value_counts().head(31))


(31,)
DOMESTIC BATTERY SIMPLE                                        1172
SIMPLE                                                          841
AGGRAVATED: OTHER DANG WEAPON                                   129
AGGRAVATED: HANDGUN                                              50
AGGRAVATED:KNIFE/CUTTING INSTR                                   49
AGGRAVATED DOMESTIC BATTERY: OTHER DANG WEAPON                   34
PRO EMP HANDS NO/MIN INJURY                                      33
AGGRAVATED DOMESTIC BATTERY: KNIFE/CUTTING INST                  31
AGG PO HANDS NO/MIN INJURY                                       28
AGGRAVATED - HANDGUN                                             20
AGG: HANDS/FIST/FEET NO/MINOR INJURY                             18
AGGRAVATED DOMESTIC BATTERY: HANDS/FIST/FEET SERIOUS INJURY      12
AGGRAVATED - OTHER DANGEROUS WEAPON                              11
AGGRAVATED DOMESTIC BATTERY - OTHER DANGEROUS WEAPON              8
AGGRAVATED - KNIFE / CUTTING INSTRUMENT   

Check the difference between DOMESTIC BATTERY SIMPLE and SIMPLE

In [74]:
battery_crime[battery_crime['description'] == 'DOMESTIC BATTERY SIMPLE'].tail(2)

unique_key case_number                  date                 block  \
9356     9412643    HW556314  2013-12-02T06:00:00Z   057XX W HIGGINS AVE   
9357     5937070    HN735766  2007-11-30T08:55:00Z  035XX S ARTESIAN AVE   

      iucr primary_type              description location_description  arrest  \
9356  0486      BATTERY  DOMESTIC BATTERY SIMPLE            APARTMENT   False   
9357  0486      BATTERY  DOMESTIC BATTERY SIMPLE               STREET   False   

      domestic  ...  ward  community_area  fbi_code  x_coordinate  \
9356     False  ...  45.0            11.0       08B     1136960.0   
9357      True  ...  12.0            59.0       08B     1160597.0   

     y_coordinate  year            updated_on   latitude  longitude  \
9356    1932689.0  2013  2018-02-10T03:50:01Z  41.971456 -87.771765   
9357    1881160.0  2007  2018-02-28T03:56:25Z  41.829598 -87.686277   

                           location  
9356  (41.971456136, -87.771764596)  
9357  (41.829598355, -87.686277105)  

[2 rows x 22 columns]

In [75]:
battery_crime[battery_crime['description'] == 'SIMPLE'].tail(2)

unique_key case_number                  date                block  iucr  \
8771    10079690    HY268175  2015-05-20T03:40:00Z  133XX S LANGLEY AVE  0460   
8772     5939953    HN738353  2007-12-02T11:40:00Z    107XX S EWING AVE  0460   

     primary_type description     location_description  arrest  domestic  ...  \
8771      BATTERY      SIMPLE  CHA PARKING LOT/GROUNDS    True     False  ...   
8772      BATTERY      SIMPLE       GROCERY FOOD STORE   False     False  ...   

      ward  community_area  fbi_code  x_coordinate y_coordinate  year  \
8771   9.0            54.0       08B     1183326.0    1817142.0  2015   
8772  10.0            52.0       08B     1202188.0    1834289.0  2007   

                updated_on   latitude  longitude  \
8771  2018-02-10T03:50:01Z  41.653426 -87.604874   
8772  2018-02-28T03:56:25Z  41.700021 -87.535279   

                           location  
8771  (41.653425905, -87.604873764)  
8772   (41.700020818, -87.53527894)  

[2 rows x 22 columns]

conclusion: The two are different since the fbi_code are different. No correlation with domestic column

# Question to answer

From this, it safe not to use description in analysis, because:
- description is messy
- Instead, how about battery crime that lead to arrest:
- how about its description?

In [86]:
battery_crime[battery_crime['arrest'] == True].head(2)

unique_key case_number                  date              block  iucr  \
151    12399489    JE271794  2021-06-19T12:01:00Z    008XX W 52ND ST  0497   
157     4719795    HM325669  2006-05-02T10:45:00Z  047XX S BISHOP ST  0484   

    primary_type                                        description  \
151      BATTERY  AGGRAVATED DOMESTIC BATTERY - OTHER DANGEROUS ...   
157      BATTERY                        PRO EMP HANDS NO/MIN INJURY   

         location_description  arrest  domestic  ...  ward  community_area  \
151                    STREET    True      True  ...  20.0            61.0   
157  SCHOOL, PUBLIC, BUILDING    True     False  ...  20.0            61.0   

     fbi_code  x_coordinate y_coordinate  year            updated_on  \
151       04B     1171616.0    1870374.0  2021  2021-09-20T04:45:04Z   
157       08B     1167467.0    1873258.0  2006  2018-02-28T03:56:25Z   

      latitude  longitude                       location  
151  41.799766 -87.646165  (41.799765636, -87.646165129)  
157  41.807770 -87.661298  (41.807769669, -87.661297945)  

[2 rows x 22 columns]

546 BATTERY crime happened to lead to arrests

In [105]:
# Check the word 'GUN' for battery crime description
battery_crime.loc[battery_crime['description'].str.contains("GUN")][40:42]

unique_key case_number                  date               block  iucr  \
4594    12217022    JD426533  2020-11-09T08:05:00Z     019XX S TROY ST  041A   
4595     9890965    HX541206  2014-12-13T08:10:00Z  012XX S ALBANY AVE  041A   

     primary_type           description location_description  arrest  \
4594      BATTERY  AGGRAVATED - HANDGUN             SIDEWALK   False   
4595      BATTERY   AGGRAVATED: HANDGUN                ALLEY   False   

      domestic  ...  ward  community_area  fbi_code  x_coordinate  \
4594     False  ...  24.0            29.0       04B     1155664.0   
4595     False  ...  24.0            29.0       04B     1155906.0   

     y_coordinate  year            updated_on   latitude  longitude  \
4594    1890335.0  2020  2020-12-17T03:44:58Z  41.854876 -87.704130   
4595    1894238.0  2014  2018-02-10T03:50:01Z  41.865582 -87.703136   

                           location  
4594  (41.854876172, -87.704129692)  
4595    (41.86558156, -87.70313629)  

[2 rows x 22 columns]

No Shooting, but mostly AGGRAVATED HANDGUN 

In [110]:
# Check where battery crime location in school or building
school_public_building = battery_crime.loc[battery_crime['location_description'].str.contains("SCHOOL|BUILDING|PUBLIC")]
battery_crime.loc[battery_crime['location_description'].str.contains("SCHOOL|BUILDING|PUBLIC")].tail(2)

unique_key case_number                  date            block  iucr  \
8812    10498608    HZ239990  2016-04-25T06:15:00Z  021XX E 87TH ST  0486   
8902    10086893    HY275780  2015-05-26T03:06:00Z  045XX W 46TH ST  0486   

     primary_type              description            location_description  \
8812      BATTERY  DOMESTIC BATTERY SIMPLE       SCHOOL, PRIVATE, BUILDING   
8902      BATTERY  DOMESTIC BATTERY SIMPLE  FACTORY/MANUFACTURING BUILDING   

      arrest  domestic  ...  ward  community_area  fbi_code  x_coordinate  \
8812    True      True  ...   8.0            45.0       08B     1191768.0   
8902    True      True  ...  14.0            57.0       08B     1147098.0   

     y_coordinate  year            updated_on   latitude  longitude  \
8812    1847705.0  2016  2018-02-10T03:50:01Z  41.737094 -87.572998   
8902    1873755.0  2015  2018-02-10T03:50:01Z  41.809546 -87.735994   

                           location  
8812  (41.737094305, -87.572998178)  
8902    (41.809546288, -87.7359941)  

[2 rows x 22 columns]

160 of battery crime happened in school or building or public

In [114]:
# Battery crime in school public building lead to arrests
school_public_building[school_public_building['arrest'] == True][24:27]

unique_key case_number                  date              block  iucr  \
7610     7889261    HT117639  2011-01-13T10:00:00Z  131XX S CARVER DR  0479   
7818     3801338    HL161728  2005-02-04T10:32:00Z   008XX E 103RD ST  0420   
7948     7484723    HS278448  2010-04-27T12:05:00Z   008XX E 103RD ST  0484   

     primary_type                          description  \
7610      BATTERY  AGG: HANDS/FIST/FEET SERIOUS INJURY   
7818      BATTERY       AGGRAVATED:KNIFE/CUTTING INSTR   
7948      BATTERY          PRO EMP HANDS NO/MIN INJURY   

          location_description  arrest  domestic  ...  ward  community_area  \
7610  SCHOOL, PUBLIC, BUILDING    True     False  ...   9.0            54.0   
7818  SCHOOL, PUBLIC, BUILDING    True     False  ...   9.0            50.0   
7948  SCHOOL, PUBLIC, BUILDING    True     False  ...   9.0            50.0   

      fbi_code  x_coordinate y_coordinate  year            updated_on  \
7610       04B     1186850.0    1818609.0  2011  2016-02-04T06:33:39Z   
7818       04B     1183988.0    1836860.0  2005  2018-02-10T03:50:01Z   
7948       08B     1183988.0    1836860.0  2010  2018-02-10T03:50:01Z   

       latitude  longitude                       location  
7610  41.657369 -87.591934   (41.65736912, -87.591933595)  
7818  41.707519 -87.601839  (41.707519433, -87.601839023)  
7948  41.707519 -87.601839  (41.707519433, -87.601839023)  

[3 rows x 22 columns]

In [119]:
battery_crime.loc[battery_crime['description'].str.contains("SHOT")]

Empty DataFrame
Columns: [unique_key, case_number, date, block, iucr, primary_type, description, location_description, arrest, domestic, beat, district, ward, community_area, fbi_code, x_coordinate, y_coordinate, year, updated_on, latitude, longitude, location]
Index: []

[0 rows x 22 columns]